modin/core/execution/dask/common/engine_wrapper.py

```diff
@@ -90,6 +91,23 @@ def deploy(
             ]
         return remote_task_future
 
+    @classmethod
+    def check_is_future(cls, item):
```

Comment:
By calling this function inside function `materialize`, you can simplify your `apply_function_on_selected_items` function by not having to do any explicit filtering.
In addition, it will be possible to simplify the code in some other places where `materialize` function is called, since before it in many situations we check the type of the object to materialize.

Reply:
As the only purpose of apply_function_on_selected_items is to filter out values based on condition `check_is_future` . If I move the check_is_future call to materialize I think  all the logic in  apply_function_on_selected_items function will move to all materialize function as in the below snippet. 

```
@classmethod
    def materialize(cls, obj_id):
        if cls.check_is_future(obj_id):
            return ray.get(obj_id)
        elif isinstance(obj_id, list):
            futures_list = []
            futures_idx = []
            for idx, item in enumerate(obj_id):
                if cls.check_is_future(item):
                    futures_idx.append(idx)
                    futures_list.append(item)
            futures_list = ray.get(futures_list)
            for idx, item in zip(futures_idx, futures_list):
                obj_id[idx] = item
            
        return obj_id
```

If I need to use this in other implementations too it would be required to repeat this logic in other materialize calls.

Reply:
I'm not sure. Let's try to move `apply_function_on_selected_items` into `_partition_mgr_cls` firstly.

Reply:
moved the logic to _partition_mgr_cls